airflow/operators/subdag_operator.py

```diff
@@ -84,9 +86,17 @@ def __init__(
 
         self.subdag = subdag
         self.executor = executor
+        self.cfg_path = None  # will be populated if running with impersonation
+
+    def pre_execute(self, context):
+        self.cfg_path = configuration_subset(MAIN_SECTIONS) if self.run_as_user else None
 
     def execute(self, context):
         ed = context['execution_date']
         self.subdag.run(
             start_date=ed, end_date=ed, donot_pickle=True,
-            executor=self.executor)
+            executor=self.executor, cfg_path=self.cfg_path)
+
+    def post_execute(self, context, result=None):
+        if self.run_as_user and self.cfg_path:
```

Comment:
Any reason not to put this in execute itself? That way you can open the file using a "with" instead which is a bit safer in terms of automatic cleanup.